imports/lib/schemas/withCommon.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from 'zod';
+import type { MongoRecordZodType } from './generateJsonSchema';
+import { TimestampFields } from './withTimestamps';
+import { UserFields } from './withUsers';
+
+// We could compose withTimestamps and withUsers, but in the case that the
+// passed in schema is not a ZodObject, we'd end up creating multiple nested
+// intersections, so this makes the resulting schema (and type) slightly (though
+// only slightly) cleaner.
+export default function withCommon<
+  T extends MongoRecordZodType
+>(schema: T):
+T extends z.ZodObject<infer Shape, infer UnknownKeys, infer Catchall> ?
+  z.ZodObject<Shape & typeof TimestampFields & typeof UserFields, UnknownKeys, Catchall> :
+  z.ZodIntersection<T, z.ZodObject<typeof TimestampFields & typeof UserFields>> {
+  return schema instanceof z.ZodObject ?
+    schema.extend({ ...TimestampFields, ...UserFields }) :
```

Comment:
Since `.extend()` can clobber fields, should we do anything to forbid `schema` from containing any of the `TimestampFields`/`UserFields` already?  If we encounter competing definitions, that's probably a bug.

I'd say this is low priority -- no need to work too hard to defend against that failure mode; it seems pretty unlikely that we'd reuse the field names.

Reply:
Yeah, that's reasonable. Added a check. (I only added it on the `ZodObject` branch because if we're intersecting in a new type, we wouldn't be clobbering any existing configuration.)